### ISSUE TITLE
Package installer fails with error message "400 Bad Request {"status":400,"message":"Unable to extract package from raw data"}"

### ISSUE THREAD

**Issue by vivekmandal** [vmware/build-tools-for-vmware-aria#208]
### Description

While pushing the vRO/vRA Package using the installer, The push fails with the below message :
Exception in thread "main" org.springframework.web.client.HttpClientErrorException: 400 Bad Request {"status":400,"message":"Unable to extract package from raw data"}

### Steps to Reproduce

1. Build and create a vRO/vRA package
2. Try installing it using installer.bat by providing the env.properties file

**Expected behavior:** The package should be pushed to the environment without any failure

**Actual behavior:** The push fails in the middle

**Reproduces how often:** It fails intermittently, I would say 50%.

**Affects Build/s:** v2.36.0

#### Client

-   Build Tools for VMware Aria Version: 2.36.0

#### Server

-   vRealize Automation Version:  8.11.2.30052 (21452955)

### Failure Logs
10:20:11.454 [main] INFO  com.vmware.pscoe.iac.artifact.VroPackageStore - Package | IMPORT | com.tsi.fci.spd.spd-core.actions-11.6.4-SNAPSHOT.package
10:20:20.641 [main] INFO  com.vmware.pscoe.iac.artifact.VroPackageStore - Package | IMPORT | com.tsi.fci.spd.spd-core.workflows-11.6.4-SNAPSHOT.package
Exception in thread "main" org.springframework.web.client.HttpClientErrorException: 400 Bad Request {"status":400,"message":"Unable to extract package from raw data"}
	at com.vmware.pscoe.iac.artifact.rest.RestClientFactory$1.handleError(RestClientFactory.java:157)
	at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:709)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:662)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:637)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:558)
	at com.vmware.pscoe.iac.artifact.rest.RestClientVro.importPackageForce(RestClientVro.java:294)
	at com.vmware.pscoe.iac.artifact.rest.RestClientVro.importPackageWithMerge(RestClientVro.java:255)
	at com.vmware.pscoe.iac.artifact.rest.RestClientVro.importPackage(RestClientVro.java:303)
	at com.vmware.pscoe.iac.artifact.VroPackageStore.importPackage(VroPackageStore.java:273)
	at com.vmware.pscoe.iac.artifact.VroPackageStore.importAllPackages(VroPackageStore.java:205)
	at com.vmware.pscoe.iac.artifact.VroPackageStore.importAllPackages(VroPackageStore.java:137)
	at com.vmware.pscoe.iac.installer.Installer.main(Installer.java:933)
#################################################################################################
########################################## -- ERROR -- ##########################################
STEP - 5. Vmware installer run 
Error - 1
Executed Command -
cmd /c .\installer.bat ..\..\spdcore-Mucf-s001_environment.properties
#################################################################################################

**Comment by Michaelpalacce** [comment]
Hey @vivekmandal, 

Can I ask for a bit more details?

1. Can you show us your environment.properties file (remove passwords and other sensitive information)
2. Do you regenerate the installer every time or does it fail from the same installer bundle?

**Comment by vivekmandal** [comment]
Hi @Michaelpalacce ,
Please find the below answers.
1. Attaching the prop file 
2. Yes. I regenerate it. It fails intermittel

[environment.properties.txt](https://github.com/vmware/build-tools-for-vmware-aria/files/13712450/environment.properties.txt)

**Comment by Michaelpalacce** [comment]
Hey @vivekmandal, the only thing I found regarding this issue on our end is it happened once because of the keystore. The signing of the package was not ok. Are you using our example keystore? Can you try using it if not?

**Comment by vivekmandal** [comment]
Hi @Michaelpalacce . Not sure how the keystore was created in the beginning. But we have been using the same keystore from a very long time. It works with the older toolchain(eg.2.25.1). somehow, now failed for the new vRBT toolchain.  If you think, its only the issue with keystore then maybe close the ticket. I would check with the team and if needed, would try to recreate the keystore. ;)

**Comment by Michaelpalacce** [comment]
Hey @vivekmandal, any luck on this one? One more thing, can you let us know which version of Node,mvn, java you are using?

**Comment by vivekmandal** [comment]
Hii @Michaelpalacce . The team decided to use the old toolchain(2.25.1) at the moment to fix the issue. Maybe we could give it a try for our next release. Regarding the environment, we are using the below versions..
Node v16.9.1
Apache Maven 3.6.3
Java version: 1.8.0_322, vendor: Temurin

**Comment by Michaelpalacce** [comment]
@vivekmandal got it, will close this for now, feel free to open if you test it

**Comment by VenelinBakalov** [comment]
@Michaelpalacce could it be because of Java? I recently had to upgrade to 11 or 17 (need to check) but this was related to the main project, I am not sure if it is affecting the installer.

**Comment by Mohammad-Makeen** [comment]
Hi. 
I'm using vRBT v2.34.0 for testing, vRBT 2.35.1 for production.

This is error is generated because when you clone repo, the cert and private_key that comes with the repo are not working.

To check if they are working or not, go to the following path in your cloned vRBT repo ...\common\keystore-example\src\main\resources
You're going to find both files as follows: private_key.pem & cert.pem. Go to [SSL Shooper](https://www.sslshopper.com/certificate-key-matcher.html) to check their validity. For my case, they are not working.

Note: I also concluded that all packages are signed using private_key & cert.

I had to generate my own keystore (cert & private_key), and I replaced them, and from scratch, I rebuild the whole vRBT.

I hope I contributed to the issue because I saw this issue when I was compiling and building vRBT for the first time.

I hope my comment was any good
